pallets/passkey/src/lib.rs

```diff
@@ -83,10 +90,40 @@ pub mod module {
 	impl<T: Config> Pallet<T> {
 		/// proxy call
 		#[pallet::call_index(0)]
-		#[pallet::weight(T::WeightInfo::proxy())]
-		pub fn proxy(_origin: OriginFor<T>) -> DispatchResult {
-			Self::deposit_event(Event::PlaceHolderEvent);
-			Ok(())
+		#[pallet::weight({
+			let dispatch_info = payload.passkey_call.call.get_dispatch_info();
+			// TODO: calculate overhead after all validations
+			let overhead = T::WeightInfo::proxy();
+			let total = overhead.saturating_add(dispatch_info.weight);
+			(total, dispatch_info.class)
+		})]
+		pub fn proxy(
+			origin: OriginFor<T>,
+			payload: PasskeyPayload<T>,
+		) -> DispatchResultWithPostInfo {
+			ensure_none(origin)?;
+			let transaction_account_id = payload.passkey_call.account_id.clone();
+			let main_origin = T::RuntimeOrigin::from(frame_system::RawOrigin::Signed(
+				transaction_account_id.clone(),
+			));
+			let result = payload.passkey_call.call.dispatch(main_origin);
+			if result.is_ok() {
+				Self::deposit_event(Event::TransactionExecutionSuccess {
+					account_id: transaction_account_id,
+				});
+			}
+			result
+		}
+	}
+
+	#[pallet::validate_unsigned]
```

Comment:
I believe it would be beneficial for us to merge into a feature branch. I’m not clear on why we are merging directly into the main branch, as we haven’t done this before. Could you please explain the reason for this deviation?